### ISSUE TITLE
DisableWebSearch doesn't work on 2004

### ISSUE THREAD

**Issue by Disassembler0** [Disassembler0/Win10-Initial-Setup-Script#300]
Microsoft has broken all GPOs and settings related to disabling Bing Search in Start Menu in 2004.

The best workaround currently is to set
```powershell
If (!(Test-Path "HKCU:\Software\Policies\Microsoft\Windows\Explorer")) {
    New-Item -Path "HKCU:\Software\Policies\Microsoft\Windows\Explorer" | Out-Null
}
Set-ItemProperty -Path "HKCU:\Software\Policies\Microsoft\Windows\Explorer" -Name "DisableSearchBoxSuggestions" -Type DWord -Value 1
```

however that disables searching also for recent entries and possibly has some other side effects.

This issue is slightly related to #262, where Microsoft managed to break web search settings in a different way.

**Comment by mariobesen** [comment]
Heyo,

To disable web search in Windows May 2020 Update (build 19041, version 2004) you must set DWORD "ConnectedSearchUseWeb" to value '0' under:
HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\Windows Search

![image](https://user-images.githubusercontent.com/45096706/80107802-d89f1f80-857b-11ea-81e1-3590ff6bb466.png)

**Comment by Disassembler0** [comment]
@mariobesen: I'm unable to replicate that. Are there any more keys/setting outside the script you have applied? Moreover, "*Don´t search the web or display web results in search*" GPO sets the `DisableWebSearch` registry key. The `ConnectedSearchUseWeb` is AFAIK obsolete since ~1803.

From what I have tried, none of the following (or combination of thereof) work:

- AllowSearchToUseLocation
- BingSearchEnabled
- ConnectedSearchUseWeb
- CortanaConsent
- DisableWebSearch

Activity history setting has also no effect.

**Comment by mariobesen** [comment]
@Disassembler0 Hm, that's interesting.

This is the default state where web search works.
![image](https://user-images.githubusercontent.com/45096706/80654218-11f3f580-8a7c-11ea-9d82-5173240b140a.png)

When I enable "Do not allow web search" policy it is not respected although "DisableWebSearch" DWORD is set to 0.
![image](https://user-images.githubusercontent.com/45096706/80654395-6b5c2480-8a7c-11ea-9a3e-beaf314ead47.png)

But, if I set "Don't search the web or display web results in Search" alongside first policy, it seems like it works:
![image](https://user-images.githubusercontent.com/45096706/80654509-af4f2980-8a7c-11ea-8ecb-5f554bd1ebc4.png)

This is on build 19041.207, Enterprise SKU.

**Comment by Disassembler0** [comment]
> This is on build 19041.207, Enterprise SKU.

Yeah, that's it. :/ Just tried in on Enterprise and it works there. On Home and Pro it doesn't.

**Comment by Kein** [comment]
Not broken, works as intended :) Microsoft need that sweet-sweet telemetry and private data.

**Comment by navossoc** [comment]
Any luck disabling the search on Windows Pro?

**Comment by rnc** [comment]
Does https://superuser.com/questions/1510739/how-to-get-rid-of-windows-10-search-web-results-triggered-from-the-start-menu help ? It seems setting `DisableSearchBoxSuggestions` DWORD32 value in `HKEY_CURRENT_USER\SOFTWARE\Policies\Microsoft\Windows\Explorer` to 1 does seem to work for me (Win 10 Home Version	10.0.19041 Build 19041 in a VM)

**Comment by navossoc** [comment]
@rnc Not really, the search are still being done on the background.
It just doesn't show the results on the start menu.

Sad shit?

**Comment by rnc** [comment]
@navossoc Ok, but at least that is less confusing to users. I wonder if the search could be blocked by the firewall?

**Comment by navossoc** [comment]
@rnc No idea, but this is definitively something I want to look into.
I don't want a keylogger on my start menu.

**Comment by Kein** [comment]
> I wonder if the search could be blocked by the firewall?

Yes and no. The rule is straightforward and you can block Search.exe which is what goes into the outer network when you type something in search. However, knowing Microsoft and their unquenchable greed for your private data, telemetry and analytics - I wouldnt be surprised some other service, that claims to be essential, mirrors the ability to upload your data. This brings us to the fact that Microsoft (intentionally) shipped broken Firewall which can't distinguish different services from each other (all native Win services are being ran through svchost container) which means you cant block specific service even if you find it - you have to block them all.

If you can live with that then this is the best solution, you gonna need to download updates manually tho and whitelist your apps you wanna use.

_(Albeit, I saw a claim that even when you are in total block mode via WFP (Windows Filtering Platform, this is what Windows Firewall runs on) with EVERYTHING explicitly blocked, there is still some network activity towards MS server/endpoints. It would be super bizzare but I cant say I'd be surprised if there are some hard-coded (at the platform code level) exception or even telemetry functionality right in the kernel. This isnt confirmed tho, I didnt have time nor currently any means to test it)_

**Comment by CHEF-KOCH** [comment]
Solution is [here](https://github.com/CHEF-KOCH/regtweaks/commit/333015f04ed91db0c1ef66d262d8f7e774320118). 

Works with all versions even Home because the firewall rule will block it since GPO is not reliable for PRO users, on Enterprise it's working via GPO).

**Comment by Kein** [comment]
This is not enough, because after an update Windows 10 will reset most of the rules MC thinks they should "manage themselves". These resets will re-enable blocked rules or re-add some deleted. What you need is to setup a Task and rework script to run on boot and iterate all rules and see if some rules that are set to block either set to to allow now or if there are duplicate rule for the same EXE/package that set to allow and purge the restored. Otherwise you end up believing you are done when in reality it was silently set to allowed and you were oblivious.

**Comment by E3V3A** [comment]
Can this be related to the #339 ?

**Comment by Moe1369** [comment]
> Can this be related to the #339 ?

Sadly I think this has to do with the Pro version afaik. I'm running Win10 Education and the provided regkey has worked. Do you by any chance run the Pro version?

**Comment by E3V3A** [comment]
> Do you by any chance run the Pro version?

Nope, using a (not updated) Home version.
